File: packages/flutter/lib/src/resources/providers.dart
/// Providers resource — `/api/providers`.
library;

import 'base.dart';

class ProvidersResource extends BaseResource {
  const ProvidersResource(super.client);

  Future<List<Map<String, dynamic>>> list() async {
    final data = await client.request('GET', 'providers');
    if (data is List) return data.whereType<Map<String, dynamic>>().toList();
    return const [];
  }

  Future<Map<String, dynamic>> get(String id) async {
    final data = await client.request('GET', 'providers/$id');
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<Map<String, dynamic>> create(Map<String, dynamic> payload) async {
    final data = await client.request('POST', 'providers', body: payload);
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<Map<String, dynamic>> update(
      String id, Map<String, dynamic> payload) async {
    final data = await client.request('PUT', 'providers/$id', body: payload);
    return (data as Map<String, dynamic>?) ?? {};
  }

  Future<void> delete(String id) =>
      client.request('DELETE', 'providers/$id');
}

File: packages/python/src/tbd_agents/resources/providers.py
from __future__ import annotations

from typing import Any, Mapping

from tbd_agents.types import Provider, ProviderCreate, ProviderUpdate

from .base import BaseResource


class ProvidersResource(BaseResource):
    def create(self, payload: ProviderCreate | Mapping[str, Any]) -> Provider:
        return self._post("/providers", payload, model=Provider)

    def list(self) -> list[Provider]:
        return self._get_list("/providers", model=Provider)

    def get(self, provider_id: str) -> Provider:
        return self._get(f"/providers/{provider_id}", model=Provider)

    def update(self, provider_id: str, payload: ProviderUpdate | Mapping[str, Any]) -> Provider:
        return self._put(f"/providers/{provider_id}", payload, model=Provider)

    def delete(self, provider_id: str) -> None:
        self._delete(f"/providers/{provider_id}")

